### PR TITLE
Update Assimp and SDL CMakeLists.txt

### PR DESCRIPTION
**Pull request by sikor666**
- Fix: https://github.com/urho3d/Urho3D/issues/2959
- Remove deprecated CMAKE_COMPILER_IS_MINGW
  https://gitlab.kitware.com/cmake/cmake/-/merge_requests/6538

### CODE REVIEW

File: Source/ThirdParty/Assimp/CMakeLists.txt [hunk 1]
```diff
@@ -106,6 +106,28 @@ INCLUDE_DIRECTORIES(
...
+  # hide all not-exported symbols
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -fvisibility=hidden -fPIC -Wall -std=c++0x")
+  SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
+  SET(LIBSTDC++_LIBRARIES -lstdc++)
+ELSEIF(MSVC)
+  # enable multi-core compilation with MSVC
+  add_compile_options(/MP)
+
+  # disable "elements of array '' will be default initialized" warning on MSVC2013
+  IF(MSVC12)
```
Comment (1vanK) [inline Source/ThirdParty/Assimp/CMakeLists.txt:120]: https://cmake.org/cmake/help/latest/variable/MSVC12.html

**Comment by 1vanK** [comment]
I don't understand what problem you are trying to solve with this bunch of flags and how they are related to MINGW

File: Source/ThirdParty/Assimp/CMakeLists.txt (hunk 1 above)
Comment (sikor666) [reply Source/ThirdParty/Assimp/CMakeLists.txt:120]: I restored the section "Grouped compiler settings" from the original file from v4.1.0 of the assimp library with some minor fixes. I will remove the rest of the conditions and leave this one for MINGW.

# Grouped compiler settings
IF ((CMAKE_C_COMPILER_ID MATCHES "GNU") AND NOT CMAKE_COMPILER_IS_MINGW)
  # hide all not-exported symbols
  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -fvisibility=hidden -fPIC -Wall -std=c++0x")
  SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
  SET(LIBSTDC++_LIBRARIES -lstdc++)
ELSEIF(MSVC)
  # enable multi-core compilation with MSVC
  add_compile_options(/MP)

  # disable "elements of array '' will be default initialized" warning on MSVC2013
  IF(MSVC12)
    add_compile_options(/wd4351)
  ENDIF()
ELSEIF ( "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" )
  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -fvisibility=hidden -fPIC -Wall -Wno-long-long -std=c++11" )
  SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
ELSEIF( CMAKE_COMPILER_IS_MINGW )
  SET( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=hidden -Wall -Wno-long-long -std=c++11" )
  SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
  ADD_DEFINITIONS( -U__STRICT_ANSI__ )
ENDIF()

**Comment by 1vanK** [comment]
Before you change anything, you need to find the cause of the problem. Because I compile with MINGW without a problem. Have you tried clean compilation after deleting CMake cache?

**Comment by sikor666** [comment]
Yes, I have clean repository before start building engine. I will add to this issue https://github.com/urho3d/Urho3D/issues/2959 script how to reproduce this problem and when you confirm i prepare fix again.

**Pull request closed without merging**